git/gitattr/tree.go

```diff
@@ -99,11 +122,10 @@ func (t *Tree) Applied(to string) []*Attr {
 		}
 	}
 
-	splits := strings.SplitN(to, "/", 2)
-	if len(splits) == 2 {
-		car, cdr := splits[0], splits[1]
-		if child, ok := t.Children[car]; ok {
-			attrs = append(attrs, child.Applied(cdr)...)
+	dirPath, remainingPath, found := strings.Cut(to, "/")
```

Comment:
This is a nice simplification!  (I'm a bit sad to lose the Lisp [terminology](https://en.wikipedia.org/wiki/CAR_and_CDR), but I think your names are going to be more helpful for most people.)